Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -48,6 +48,7 @@ class TabLocationView: UIView, FeatureFlaggable {
     private let menuBadge = BadgeWithBackdrop(imageName: ImageIdentifiers.menuBadge, backdropCircleSize: 32)
 
     var url: URL? {
+        willSet { handleShoppingAdsCacheURLChange(newURL: newValue) }
```

Comment:
Is this the source of truth from where we know the url changed?

Reply:
It is always in sync whenever the url changes.